### PR TITLE
Pass components v2 in flow calls

### PR DESCRIPTION
**Pull request by orius123**
Fixes https://github.com/descope/etc/issues/3680

### CODE REVIEW

**Comment by orius123** [comment]
Really hard to know the state when all tests fail like that

File: packages/web-component/src/lib/types.ts [hunk 1]
```diff
@@ -135,3 +131,35 @@ export interface ILogger {
...
+
+export type FlowConfig = {
+  startScreenId?: string;
+  version: number;
+  targetLocales?: string[];
+  conditions?: ClientCondition[];
+  condition?: ClientCondition;
+};
+
+export interface Configuration {
```
Comment (asafshen) [inline packages/web-component/src/lib/types.ts:156]:
```suggestion
export interface ProjectFlowsConfiguration {
```

File: packages/web-component/src/lib/descope-wc/DescopeWc.ts [hunk 2]
```diff
@@ -656,8 +674,15 @@ class DescopeWc extends BaseDescopeWc {
         `[${ELEMENT_TYPE_ATTRIBUTE}="polling"]`
       );
       if (loader) {
+        const flowConfig = await this.getFlowConfig();
+        const projectConfig = await this.getProjectConfig();
```
Comment (asafshen) [inline packages/web-component/src/lib/descope-wc/DescopeWc.ts:678]: I would not let the caller set this here

instead, we can pass it where the function is defined 
(also true for the [change bellow](https://github.com/descope/descope-js/pull/255/files#diff-4e98988585b972daa77cdd3c14af366d7021a2ef1666a77d6b95f50fc68e60c9R768)
image)

**Review by asafshen — COMMENTED** [review]
looks good!

noting that maybe, instead passing version and componentVersion everywhere that we want to call `start/`next`, we can set it once, so the callers won't have to fetch it every time 

this requires a bit of design, but I think it will worth it

File: packages/web-component/src/lib/descope-wc/DescopeWc.ts (hunk 2 above)
Comment (orius123) [reply packages/web-component/src/lib/descope-wc/DescopeWc.ts:678]: Talked to @nirgur, we prefer to do the types changes after fixing all the tests in the v2 branch

**Comment by orius123** [comment]
Waiting for tests and will merge after

**Pull request merged**